dotnet/src/webdriver/DriverService.cs

```diff
@@ -363,23 +367,29 @@ private void Stop()
                 {
                     Uri shutdownUrl = new Uri(this.ServiceUrl, "/shutdown");
                     DateTime timeout = DateTime.Now.Add(this.TerminationTimeout);
-                    while (this.IsRunning && DateTime.Now < timeout)
+                    using (var httpClient = new HttpClient())
                     {
-                        try
-                        {
-                            // Issue the shutdown HTTP request, then wait a short while for
-                            // the process to have exited. If the process hasn't yet exited,
-                            // we'll retry. We wait for exit here, since catching the exception
-                            // for a failed HTTP request due to a closed socket is particularly
-                            // expensive.
-                            HttpWebRequest request = HttpWebRequest.Create(shutdownUrl) as HttpWebRequest;
-                            request.KeepAlive = false;
-                            HttpWebResponse response = request.GetResponse() as HttpWebResponse;
-                            response.Close();
-                            this.driverServiceProcess.WaitForExit(3000);
-                        }
-                        catch (WebException)
+                        httpClient.DefaultRequestHeaders.ConnectionClose = true;
```

Comment:
What if we set Timeout (5 seconds?) explicitly?

Reply:
`this.TerminationTimeout` is set to 10 seconds and is configurable. Why explicit 5 seconds?

Reply:
So, leaving the logic as it was before.